unit_tests/src/test_resource_administration.cpp

```diff
@@ -77,6 +80,33 @@ TEST_CASE("resource administration")
         REQUIRE(info->host_name() == ufs_info.host_name);
     }
 
+    // see issue #5170
+    SECTION("verify resource id from json serialization")
+    {
+        // we need a fresh connection becuase the cache is probably out of date
+        auto conn_pool = irods::make_connection_pool();
+        auto conn = conn_pool->get_connection();
+
+        auto [ec, info] = adm::client::resource_info(conn, ufs_info.resource_name);
+        REQUIRE(ec.value() == 0);
+        REQUIRE(info->type() == ufs_info.resource_type);
+
+        bytesBuf_t *jbuf;
```

Comment:
prefer initializing your variables (unless you can make certain guarantees).  right now, this could point to anything in memory and cause the program to crash.

you can always use curly braces to zero out primitive types and C structures.  for example:
```c++
bytesBuf_t* bbuf{};  // Equivalent to: bytesBuf_t* bbuf = nullptr;
```